### ISSUE TITLE
SDXL checkpoint bad result?

### ISSUE THREAD

**Issue by lingcong-k** [TencentARC/BrushNet#43]
### Describe the bug

I use test_brushnet_sdxl.py and use the segmentation sdxl checkpoint on the cake images in the example folder. but i got very bad results.
am I doing sth wrong?
"A cake on the table."
![image](https://github.com/TencentARC/BrushNet/assets/73646719/e9cb3392-0230-4553-874e-1edeec3a120a)
![image](https://github.com/TencentARC/BrushNet/assets/73646719/9bd15052-22eb-4a17-b1b5-169331ad2f4c)

result
![image](https://github.com/TencentARC/BrushNet/assets/73646719/74cc4219-a129-4115-8056-6bece29bb5bd)

### Reproduction

run test_brushnet_sdxl.py. but with stable-diffusion-xl-base-1.0 checkpoint and segmentation_mask_brushnet_ckpt_sdxl_v0

### System Info

A100

**Comment by Shuvo001** [comment]
@lingcong-k you can try to generate the outputs with ComfyUI. SDXL version outputs are not as bad as yours

**Comment by lawsonxwl** [comment]
为什么用 test_brushnet_sdxl.py 输出的是inpaint背景的结果？ 我这边用这个代码和图片输出的是inpaint前景的结果。。。

**Comment by kjzju** [comment]
> ### Describe the bug
> I use test_brushnet_sdxl.py and use the segmentation sdxl checkpoint on the cake images in the example folder. but i got very bad results. am I doing sth wrong? "A cake on the table." ![image](https://private-user-images.githubusercontent.com/73646719/337203538-e9cb3392-0230-4553-874e-1edeec3a120a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE2MjAyOTAsIm5iZiI6MTcyMTYxOTk5MCwicGF0aCI6Ii83MzY0NjcxOS8zMzcyMDM1MzgtZTljYjMzOTItMDIzMC00NTUzLTg3NGUtMWVkZWVjM2ExMjBhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIyVDAzNDYzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNlZjJjNTIxYTZjM2U2MTAzNDRlYTBiYzVhNGNhZWM3MjY0YjM4MWI4ZTJkNTQzMGI3MjM5ZTBiM2NhMmJjMTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XuYM3jL5tcuHIOuD1OV74qvPeSNdSWWKp48U9mNo5LY) ![image](https://private-user-images.githubusercontent.com/73646719/337203693-9bd15052-22eb-4a17-b1b5-169331ad2f4c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE2MjAyOTAsIm5iZiI6MTcyMTYxOTk5MCwicGF0aCI6Ii83MzY0NjcxOS8zMzcyMDM2OTMtOWJkMTUwNTItMjJlYi00YTE3LWIxYjUtMTY5MzMxYWQyZjRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIyVDAzNDYzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1ZmM4NTYwMTg3NTg3ZTkzYTYxZGQ3YzE2M2Q4MDQyOGU0MTM5Yzc1ZTYzM2NhOGQ0ZGQyYzUwY2E2YTc0MjgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.UAtu_e7B76OyxzgpVcQNQhLFYXKOGLNXhx-i_uCV1MQ)
> 
> result ![image](https://private-user-images.githubusercontent.com/73646719/337203440-74cc4219-a129-4115-8056-6bece29bb5bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE2MjAyOTAsIm5iZiI6MTcyMTYxOTk5MCwicGF0aCI6Ii83MzY0NjcxOS8zMzcyMDM0NDAtNzRjYzQyMTktYTEyOS00MTE1LTgwNTYtNmJlY2UyOWJiNWJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIyVDAzNDYzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyYTA3ZmRiYmY0Y2Q2YWYyYmU1YzQ4NmRlYWMwYzY3NGMxMjUzOThjN2RlYzY0OWRjZGRmMDQ1NjY0Yjg2NzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fpTb7Et9DWLqYeSJq4p02WCELi5b-TMbWYcQVMLWTds)
> 
> ### Reproduction
> run test_brushnet_sdxl.py. but with stable-diffusion-xl-base-1.0 checkpoint and segmentation_mask_brushnet_ckpt_sdxl_v0
> 
> ### Logs
> _No response_
> 
> ### System Info
> A100
> 
> ### Who can help?
> _No response_

Have you figure out the problem and fix it?
